---
title: "Fig.4bc"
author: "ZH"
date: "2024-08-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Fig.4a 
```{r}
library(readxl)
library(tidyverse)
library(ape)
library(circlize)

# loading data
dataAtLSPHERE <- read.delim("isolates_taxonomy.txt", header = TRUE, strip.white = TRUE, stringsAsFactors = FALSE)
dataAtLSPHERE$Well1 <- gsub("_","",dataAtLSPHERE$Well1)
dataAtLSPHERE <- dataAtLSPHERE %>% dplyr::select(Well1, Phylum, Order, Family, Genus)

# color assign for Phylum 
colorScaleStrains <- setNames(
  c("#fd302f","#CD2990","#00FF00","#fdf399","#EDB31E",
    "#a5d3ed","#4682B4","#B5B5B5","#FF69B4"),  
  c("Alphaproteobacteria","Betaproteobacteria","Gammaproteobacteria","Actinobacteria",
    "Firmicutes","Verrucomicrobia", "Bacteroidetes","Flavobacteria","Myxococcota"))

# define the color for each strain according their phylum information 
annot_color_Leaf_colNr <- function(strain, df = dataAtLSPHERE){
  strain = as.character(strain)
  if(strain %in% df[["Well1"]]){
    phylum = as.character(df[["Phylum"]][df[["Well1"]] == strain])
    if (phylum == "Alphaproteobacteria") {
      color <- "#fd302f"
    } else if (phylum == "Betaproteobacteria") {
      color <- "#CD2990" 
    } else if (phylum == "Gammaproteobacteria") {
      color <- "#00FF00"
    } else if (phylum == "Actinobacteria") {  
      color <- "#fdf399"
    } else if (phylum == "Firmicutes") {
      color <- "#EDB31E"
    } else if (phylum == "Verrucomicrobia") {
      color <- "#a5d3ed"
    } else if (phylum == "Bacteroidetes") {
      color <- "#4682B4"
    } else if (phylum == "Flavobacteria") {
      color <- "#B5B5B5"
    } else if (phylum == "Myxococcota") {
      color <- "#FF69B4"
    } else  { 
      color <- "grey"
    }
  } 
  return(color)
}

# loading the phylogenetic tree 
tree2plot <- read.tree("RH_all_da_rep3.nwk")

#Calculation of Plotting Parameters
angle <- (2 * pi) / length(tree2plot$tip.label)
angles <- ((1:length(tree2plot$tip.label)) - 1) * angle
names(angles) <- tree2plot$tip.label
start.angles <- (angles - angle / 2) * (180 / pi)
end.angles <- (angles + angle / 2) * (180 / pi)
radius <- 0.7

# gain of colors of strain
cols_strains <- sapply(tree2plot$tip.label, annot_color_Leaf_colNr)

# Setting the branch length of phylogenetic tree  
tree2plot$edge.length <- rep(0.029, length(tree2plot$edge.length))
colorScaleStrains2 <- paste(colorScaleStrains, "86", sep = "")
names(colorScaleStrains2) <- names(colorScaleStrains) 
colorScaleStrains2


# plotting the phylogenetic tree 
pdf("fig.pdf", height = 5, width = 5)
par(oma = c(0, 0, 0, 0))

Here's the simplified code with English comments:

# Reduce white space
plot(tree2plot, type = "fan", align.tip.label = TRUE, plot = FALSE, no.margin = TRUE, x.lim = c(-0.8, 0.8), y.lim = c(-0.8, 0.8)) 

# Draw background sectors with different colors representing different clades
for (i in unique(tree2plot$tip.label)) {
  # Get the color for the current strain
  col <- cols_strains[i]
  
  # Check if the color is valid
  if (!is.na(col)) {
    # If the color is valid, add transparency
    col <- paste(col, "30", sep = "") 
  } else {
    # If the color is invalid, use default transparent grey
    col <- "grey30"  
  }
  
  draw.sector(start.angles[i], end.angles[i], rou1 = 0.63, 
              col = col, clock.wise = FALSE, border = NA)
}
par(new = TRUE)

# Redraw the tree plot to ensure it is plotted above the sectors
plot(tree2plot, type = "fan", align.tip.label = TRUE, 
     x.lim = c(-0.8, 0.8), y.lim = c(-0.8, 0.8), 
     edge.width = 0.8,
     cex = 0.4, label.offset = 0.01)  

# Add legend
legend(x = 0.5,y = 1.2, legend = names(colorScaleStrains2), fill = colorScaleStrains2, cex = 0.3,border = NA)

# Add outer circle with colors corresponding to the sectors
outer_radius <- 0.67  
for (i in unique(tree2plot$tip.label)) {
  outer_col <- adjustcolor(cols_strains[i], alpha.f = 0.5)  
  draw.sector(start.angles[i], end.angles[i], rou1 = outer_radius - 0.02, rou2 = outer_radius, col = outer_col, clock.wise = FALSE, border = NA)
}

# Customization for adding multiple outer circles.

dev.off()
```


# Fig.4b.c 
```{r}
# loading the required packages
library(dplyr)
library(allparametertest)  
library(ggplot2)
library(ggpubr)

# loading soure data
sj <- read.csv("25PEG.csv", header = TRUE)
sj$isolates <- ifelse(sj$isolates == "1.00E+05", "1E5", sj$isolates)

# parameter summary 
e1 <- summarySE(sj, groupvars = c("treat", "isolates"), measurevar = "freshweight")
e2 <- summarySE(sj, groupvars = c("treat", "isolates"), measurevar = "PRL")

# factors seeting
isolates_levels <- rev(c("DER","4F10","AR","gl2","1E5","2D3","22D6","10D7","5A12","10C12","18F2","25F3","21B8","Buffer"))
bardatai <- sj %>%
  mutate(
    isolates = factor(isolates, levels = isolates_levels),
    treat = factor(treat, levels = c("RM", "RP"))
  )

# freshweight 
p0 <- ggplot(bardatai, aes(x = isolates, y = freshweight, fill = as.factor(treat))) +
  geom_boxplot(
    position = position_dodge(0.85),
    width = 1.0,
    outlier.shape = NA
  ) +
  geom_jitter(
    position = position_jitterdodge(jitter.width = 0.4, dodge.width = 0.85),
    aes(color = treat),
    color = 'black',
    size = 1.0,
    shape = 21,
    stroke = 1
  ) +
  stat_compare_means(
    method = "t.test",
    aes(label = ..p.signif..),
    label.x.npc = "center",
    label.y = 70,
    size = 3.0
  ) +
  labs(x = "Genotypes", y = "Shoot fresh weight") +
  scale_fill_manual(values = c("#04ae54", "#70A5D9"), guide = guide_legend(reverse = TRUE)) +
  facet_grid(~isolates, scales = 'free') +
  theme_classic()

p0

#----PRL
p1 <- ggplot(bardatai, aes(x = isolates, y = PRL, fill = as.factor(treat))) +
  geom_boxplot(
    position = position_dodge(0.85),
    width = 1.0,
    outlier.shape = NA
  ) +
  geom_jitter(
    position = position_jitterdodge(jitter.width = 0.4, dodge.width = 0.85),
    aes(color = treat),
    color = 'black',
    size = 1.0,
    shape = 21,
    stroke = 1
  ) +
  stat_compare_means(
    method = "t.test",
    aes(label = ..p.signif..),
    label.x.npc = "center",
    label.y = 10,
    size = 3.0
  ) +
  labs(x = "Genotypes", y = "Primary root length (cm)") +
  scale_fill_manual(values = c("#04ae54", "#70A5D9"), guide = guide_legend(reverse = TRUE)) +
  facet_grid(~isolates, scales = 'free') +
  theme_classic()
p1
ggarrange(p0,p1,ncol = 1,nrow = 2,labels = c("b","c"))
```

### Fig.4bc显著性标记
```{r}
# 加载必要的库
library(dplyr)
library(agricolae)

# 读取并预处理数据
process_data <- function(filepath) {
  data <- read.csv(filepath, header = TRUE)
  data$isolates <- ifelse(data$isolates == "1.00E+05", "1E5", data$isolates)
  return(data)
}

# 分析函数
analyze <- function(data, treat, parameter) {
  subset_data <- data %>% filter(treat == !!treat)
  subset_data$parameter <- subset_data[[parameter]]
  subset_data <- subset_data %>% select(parameter, isolates)
  
  fit <- aov(parameter ~ isolates, subset_data)
  print(TukeyHSD(fit))
  stat_res2 <- LSD.test(fit, "isolates", p.adj = "bonferroni")
  print(stat_res2)
  
  name <- paste0(parameter, unique(subset_data$treat)[1])
  print(name)
}

# 主程序
filepath <- "25PEG.csv"
data <- process_data(filepath)

# 分析 RP 下的 freshweight 和 PRL
analyze(data, "RP", "freshweight")
analyze(data, "RP", "PRL")

# 分析 RM 下的 freshweight 和 PRL
analyze(data, "RM", "freshweight")
analyze(data, "RM", "PRL")

# 和buffer 没有相同字母的标记 * 号
```
